packages/eslint-plugin-svelte/src/utils/svelte-context.ts

```diff
@@ -0,0 +1,201 @@
+import type { RuleContext } from '../types.js';
+import fs from 'fs';
+import path from 'path';
+import { getPackageJson } from './get-package-json.js';
+import { getFilename, getSourceCode } from './compat.js';
+
+const isRunInBrowser = !fs.readFileSync;
+
+export type SvelteContext = {
+	svelteVersion: '3/4' | 5;
```

Comment:
This is a union of a string literal and a number literal - they should both be the same type to make it less confusing for users.

Reply:
But the design is good!

Reply:
I think we can recognize this automatically via TypeScript? I don't have strong opinion for this.

Reply:
Yes, we can, but not everyone uses TS and I think this is confusing anyway... I'd just make them all strings...

Reply:
I think this is also just personal preference. I find treating numbers as strings can cause confusion, so I’ve always handled numbers as numbers. That said, a union like `"3/4" | 5` can be validated by TypeScript’s type checking, so I don’t see it as a problem. (However, if I'm a reviewer and someone submitted a PR with `"3/4" | "5"`, I wouldn’t object to it.)

Reply:
I noticed that we’ve repeated the exchange twice. I respect your opinion.

https://github.com/sveltejs/eslint-plugin-svelte/pull/980/commits/531911a246216a0fcbb6993ccb4a12576b01eb7c